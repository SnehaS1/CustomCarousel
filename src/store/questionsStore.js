import { create } from 'zustand';

const questionsList = [{
  id: 0,
  questions: 'How was your overall week ?',
  rating: null,
  backgroundColor: '#BFB2F3'
}, {
  id: 1,
  questions: 'How would you rate your satisfaction with the customer service representative you spoke to?',
  rating: null,
  backgroundColor: '#96CAF7'
}, {
  id: 2,
  questions: 'How would you rate the effort you had to make to get in touch with us?',
  rating: null,
  backgroundColor: '#9CDCAA'
}, {
  id: 3,
  questions: 'How would you rate the effort you had to make to speak directly to someone on the customer service team?',
  rating: null,
  backgroundColor: '#E5E1AB'
}, {
  id: 4,
  questions: 'How would you rate the quality of your customer service experience ?',
  rating: null,
  backgroundColor: '#F8A3A8'
}, {
  id: 5,
  questionsListFinal: [],
  ratings: [],
  backgroundColor: '#ff80bf'
}];

const handleFeedback = (val) => {
  switch (val) {
    case 0:
      return 'Nuetral';
    case 1:
      return 'Like';
    case -1:
      return 'Dislike';
  }
};

export const useQuestionsStore = create((set) => ({
  // initial state
  questionsList: questionsList,
  activeIndex: 0,
  // methods for anipulating state
  setActiveindex: (id) => {
    set(() => ({ activeIndex: id }));
  },
  updateQuestions: (id, val) => {

    set((state) => ({
      questionsList: state.questionsList.map((question) =>
        question.id === id
          ? ({ ...question, rating: handleFeedback(val) })
          : question
      ),
    }));



  },
}));