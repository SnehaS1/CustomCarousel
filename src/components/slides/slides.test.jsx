import { act, cleanup, render, renderHook, screen, within } from "@testing-library/react";
import { useQuestionsStore } from "../../store/questionsStore";
import userEvent from '@testing-library/user-event'
import SlideComponent from "./slides";

describe("Slide Navigator component renders", () => {
    beforeEach(() => {
        const { result } = renderHook(() => useQuestionsStore());
        // Reset the state
        act(() => {
            result.current.activeIndex = 0;
        });
    });
    afterEach(cleanup);

    it("matches snapshot", () => {
        const { asFragment } = render(<SlideComponent />);
        expect(asFragment()).toMatchSnapshot();
    });
    it("should render Slides Componnet Emoji correctly", () => {
        renderSlideNavigator();
        const { result } = renderHook(() => useQuestionsStore());
        let { activeIndex, questionsList, setActiveindex } = result.current;
        const element = screen.getByRole('slide-panel');
        expect(element).toBeInTheDocument();

    });
    it("should render Slides Componnet Table correctly", () => {
        renderSlideNavigator();
        const { result } = renderHook(() => useQuestionsStore());
        let { activeIndex, questionsList, setActiveindex } = result.current;
        act(() => {
            result.current.setActiveindex(2);
        });
        let questionPart = questionsList.slice(0, -1).find(o => !!o.rating);
        // const element = screen.getByTestId('hide-feedback');

        // expect(element).toBeInTheDocument();

    });
})

const renderSlideNavigator = () => {
    return render(<SlideComponent />)
};