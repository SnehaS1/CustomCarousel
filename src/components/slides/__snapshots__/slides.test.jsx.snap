// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide Navigator component renders matches snapshot 1`] = `
<DocumentFragment>
  <aside
    class="panel"
    role="slide-panel"
  >
    <section
      class="slide-container"
      id="question-container"
    >
      <h2
        class="panel-question"
        style="animation: animation: 2s anim-lineUp ease-out 1;"
      >
        How was your overall week ?
      </h2>
    </section>
    <section
      class="slide-container"
      id="emoji-container"
      style="background-color: white;"
    >
      <section>
        <ul
          class="emoji-list"
          role="feedback-section"
        >
          <li
            data-testid="emoji-list-li"
            role="emoji"
          >
            <p
              class="emoji-wrapper"
              data-testid="emoji-wrapper"
            >
              <span
                class="emoji"
              >
                👍
              </span>
              <span
                class="tooltip"
                data-testid="emoji-tooltip"
              >
                Like
              </span>
            </p>
          </li>
          <li
            data-testid="emoji-list-li"
            role="emoji"
          >
            <p
              class="emoji-wrapper"
              data-testid="emoji-wrapper"
            >
              <span
                class="emoji"
              >
                🤔
              </span>
              <span
                class="tooltip"
                data-testid="emoji-tooltip"
              >
                Nuetral
              </span>
            </p>
          </li>
          <li
            data-testid="emoji-list-li"
            role="emoji"
          >
            <p
              class="emoji-wrapper"
              data-testid="emoji-wrapper"
            >
              <span
                class="emoji"
              >
                👎
              </span>
              <span
                class="tooltip"
                data-testid="emoji-tooltip"
              >
                Dislike
              </span>
            </p>
          </li>
        </ul>
      </section>
    </section>
  </aside>
</DocumentFragment>
`;
